code/components/gta-net-five/src/MumbleVoice.cpp

```diff
@@ -832,6 +841,23 @@ static HookFunction hookFunction([]()
 			context.SetResult<int>(channelId);
 		});
 
+		fx::ScriptEngine::RegisterNativeHandler("MUMBLE_IS_CONNECTED", [](fx::ScriptContext& context)
+		{
+			context.SetResult<bool>(g_mumble.connected ? true : false);
+		});
+		
+		
+		fx::ScriptEngine::RegisterNativeHandler("MUMBLE_SET_SERVER_ADDRESS", [](fx::ScriptContext& context)
+		{
+			auto address = context.GetArgument<const char*>(0);
+			int port = context.GetArgument<int>(1);
+
+			g_mumble.overrideAddress = address;
```

Comment:
Do net::PeerAddress resolution here _and check if the std::optional is valid_ before trying to dereference/use .get(). If it's not valid, `throw std::exception("Couldn't resolve Mumble server address.")` or so to propagate to script.

This thing you're doing now will lead to a nasty _deferred_ crash if the address is invalid.